### PR TITLE
test(map): add subscription assertions on map tests

### PR DESCRIPTION
**Pull request by staltz**

### CODE REVIEW

File: spec/operators/map-spec.js [hunk 1]
```diff
@@ -88,10 +116,12 @@ describe('Observable.prototype.map()', function () {
     });
 
     expectObservable(r).toBe(expected, values);
+    expectSubscriptions(a.subscriptions).toBe(asubs);
   });
 
   it('should map with index until completed', function () {
     var a = hot('-5-^-4--3---2----1--|--8--|--#');
```
Comment (kwonoj) [inline spec/operators/map-spec.js]: (I know this is out of scope for PR) do we allow this kind of marble diagram? maybe this need to be refactored in further too?

Comment (staltz) [inline spec/operators/map-spec.js]: AFAIK RxJS 4 tests do this kind of thing too. I think it's testing whether the Subject behaves well even though we are pushing it beyond its limits, asking it to do what it cannot do.

Comment (kwonoj) [inline spec/operators/map-spec.js]: Yes, some of RxJS 4 test does and while migrated into RxJS 5 some of them were removed those (emit-after-completion). Got curious whether we'll apply same consistency here..

Comment (staltz) [inline spec/operators/map-spec.js]: Yeah IMO it's enough if the Subject spec does this test for emission after completion and operator specs like this file don't.

**Comment by kwonoj** [comment]
Merged with 417dc0a, thanks @staltz

**Pull request closed without merging**